jaeger_client/codecs.py

```diff
@@ -232,3 +235,50 @@ def extract(self, carrier):
         return SpanContext(trace_id=trace_id, span_id=span_id,
                            parent_id=parent_id, flags=flags,
                            baggage=None)
+
+
+def header_to_hex(header):
+    if not isinstance(header, (str, unicode)):
+        return None
+    return int(header, 16)
+
+
+class B3Codec(Codec):
+    """
+    Support B3 header properties
+
+    """
+    def __init__(self):
+        self.trace_header = 'X-B3-TraceId'
+        self.span_header = 'X-B3-SpanId'
+        self.parent_span_header = 'X-B3-ParentSpanId'
+        self.sampled_header = 'X-B3-Sampled'
+        self.flags_header = 'X-B3-Flags'
+
+    def inject(self, span_context, carrier):
+        if not isinstance(carrier, dict):
+            raise InvalidCarrierException('carrier not a dictionary')
+        carrier[self.trace_header] = format(span_context.trace_id, 'x')
+        carrier[self.span_header] = format(span_context.span_id, 'x')
+        if span_context.parent_id is not None:
+            carrier[self.parent_span_header] = format(span_context.parent_id, 'x')
+        carrier[self.flags_header] = str(span_context.flags)
+
+    def extract(self, carrier):
+        if not isinstance(carrier, dict):
+            raise InvalidCarrierException('carrier not a dictionary')
+        trace_id = header_to_hex(carrier.get(self.trace_header.lower()))
```

Comment:
a. for better performance `lower()` could've been called in the constructor
b. if there actually a convention among python http frameworks to represent headers as a lower-case dict? Our main codec makes case-insensitive comparisons

Reply:
The popular Python http frameworks that I know all do `.lower()` on all http headers. I think that this is the most common approach for case insensitive comparisons (vs `.upper()`)

Reply:
ok, well we can start with this assumption. Perhaps add it to the readme. Which reminds me - please add/update a section in the readme documenting how the B3 codec can be configured.